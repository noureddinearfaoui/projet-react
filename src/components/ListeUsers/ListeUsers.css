
.table-title {
    padding-bottom: 15px;
    background: #435d7d;
    color: #fff;
    padding: 16px 30px;
   
    border-radius: 3px 3px 0 0;
}
#btnshow
{
    color: #fff;
    margin-right:1px ;
    
    }

.btnPartie
{
    display: flex;
    flex-direction: row;
    justify-content:space-between;
}
.headerContenu
{
    border-color: #e9e9e9;
    padding: 12px 15px;
    vertical-align: middle;
    border-bottom: 2px solid #ddd;
    display: flex;
    flex-direction: row;
    justify-content:space-between;
    color: #566787;
    background: #f5f5f5;
    font-family: 'Varela Round', sans-serif;
    font-size: 13px;
    
}
.rowElement
{
    border-color: #e9e9e9;
    padding: 12px 15px;
    vertical-align: middle;
    border-bottom: 1px solid #ddd;
    display: flex;
    flex-direction: row;
    justify-content:space-between;
    color: #566787;
    background: #fff;
    font-family: 'Varela Round', sans-serif;
    font-size: 13px;
    
}
.headerContenu p{
    font-size: 20px;
    font-family: Georgia, serif;
}
.icons
{
    font-size: 25px;
}
.iconsOpr
{
    display: flex;
    flex-direction: row;
    justify-content:space-between;
    margin-right: 1px;
}
.iconEdit
{
    color:#FFC107;
}
.iconDelete
{
    color:#F44336;
}
.Toast
{
    position: 'absolute';
      top: 0;
      right: 0;
      background: #28a745;
}