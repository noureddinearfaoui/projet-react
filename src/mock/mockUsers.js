export default [
  {
    id: "2",
    nom: "marwen",
    prenom: "hanzouli",
    email: "marwen",
    password: "hanzouli",
    adresse: "beginner",
    tel: "25995310",
    dateNaissance: "2020-04-11",
    role: "user",
    banni: false,
    image: require("../components/images/m1.png"),
  },
  /*{
    id: "3",
    nom: "noureddine",
    prenom: "arfaoui",
    email: "noureddine",
    password: "arfa",
    adresse: "beginner",
    tel: "93985040",
    dateNaissance: "2020-04-11",
    role: "user",
    banni: false,
    image: false,
  },
  {
    id: "4",
    nom: "Amal",
    prenom: "BouAbdallah",
    email: "amal",
    password: "amal",
    adresse: "beginner",
    tel: "52049207",
    dateNaissance: "2020-04-11",
    role: "user",
    banni: false,
    image: require("../components/images/a1.jpg"),
  },*/
];
