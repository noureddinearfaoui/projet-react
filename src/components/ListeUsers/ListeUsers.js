import React, { useState, memo } from "react";

import { ToastsContainer, ToastsStore } from "react-toasts";
import "./ListeUsers.css";
import User from "../user/User";

function ListeUsers({ users, banni }) {
  const toast = (msg = null) => {
    if (msg === null) ToastsStore.success("Hey, success!");
    else ToastsStore.success("Hey!" + msg);
  };
  const toastError = (msg = null) => {
    if (msg === null) ToastsStore.error("Hey, error!");
    else ToastsStore.error("Hey!" + msg);
  };

  //const userAuth = JSON.parse(localStorage.getItem('user'));

  return (
    <div className="Livreliste">
      <ToastsContainer store={ToastsStore} />
      <div className="table-title">
        <div className="row">
          <div className="col-lg-9 col-md-9 col-sm-12"></div>
        </div>
      </div>
      <div className="contenu">
        <div className="headerContenu">
          <div>nom</div>
          <div>prenom</div>
          <div>Emprunts</div>
          <div>Banni</div>
        </div>
        <div className="BodyContenu">
          {users.map((user) => (
            <User key={user._id} user={user} users={users} banni={banni} />
          ))}
        </div>
      </div>
    </div>
  );
}

export default memo(ListeUsers);
