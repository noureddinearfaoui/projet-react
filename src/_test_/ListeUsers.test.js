import React from "react";
import { render } from "@testing-library/react";
import ListeUsers from "./../components/ListeUsers/ListeUsers";
//import user from "@testing-library/user-event";
import mockUsers from "../mock/mockUsers";
import MutationObserver from "mutation-observer";
global.MutationObserver = MutationObserver;

describe("test users list", () => {
  test("should render Liste of users", () => {
    const mockbanni = "false";
    const UsersComponent = render(
      <ListeUsers banni={mockbanni} users={mockUsers} />
    );
    expect(UsersComponent).toMatchSnapshot();
    // debug()
  });
});
