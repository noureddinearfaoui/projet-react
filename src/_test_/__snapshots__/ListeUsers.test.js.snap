// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test users list should render Liste of users 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Livreliste"
      >
        <div
          class="table-title"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-9 col-md-9 col-sm-12"
            />
          </div>
        </div>
        <div
          class="contenu"
        >
          <div
            class="headerContenu"
          >
            <div>
              nom
            </div>
            <div>
              prenom
            </div>
            <div>
              Emprunts
            </div>
            <div>
              Banni
            </div>
          </div>
          <div
            class="BodyContenu"
          >
            <div
              class="User"
            >
              <div
                class="rowElement false"
              >
                <div>
                  marwen
                </div>
                <div>
                  hanzouli
                </div>
                <div>
                  <div
                    class="EmpruntsDetails"
                  >
                    <span
                      class="icons "
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 512 512"
                        width="1em"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M255.8 112c-80.4 0-143.8 50.6-219.6 133.3-5.5 6.1-5.6 15.2-.1 21.3C101 338.3 158.2 400 255.8 400c96.4 0 168.7-77.7 220.1-134 5.3-5.8 5.6-14.6.5-20.7C424 181.8 351.5 112 255.8 112zm4.4 233.9c-53 2.4-96.6-41.2-94.1-94.1 2.1-46.2 39.5-83.6 85.7-85.7 53-2.4 96.6 41.2 94.1 94.1-2.1 46.2-39.5 83.6-85.7 85.7z"
                        />
                        <path
                          d="M256 209c0-6 1.1-11.7 3.1-16.9-1 0-2-.1-3.1-.1-36.9 0-66.6 31.4-63.8 68.9 2.4 31.3 27.6 56.5 58.9 58.9 37.5 2.8 68.9-26.9 68.9-63.8 0-1.3-.1-2.6-.1-3.9-5.6 2.5-11.7 3.9-18.2 3.9-25.2 0-45.7-21.1-45.7-47z"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div>
                  <svg
                    class="add iconbani"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 1024 1024"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM704 536c0 4.4-3.6 8-8 8H544v152c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V544H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h152V328c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v152h152c4.4 0 8 3.6 8 8v48z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="toasts-container "
      style="bottom: 10px; right: 10px;"
    />
  </body>,
  "container": <div>
    <div
      class="Livreliste"
    >
      <div
        class="table-title"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-9 col-md-9 col-sm-12"
          />
        </div>
      </div>
      <div
        class="contenu"
      >
        <div
          class="headerContenu"
        >
          <div>
            nom
          </div>
          <div>
            prenom
          </div>
          <div>
            Emprunts
          </div>
          <div>
            Banni
          </div>
        </div>
        <div
          class="BodyContenu"
        >
          <div
            class="User"
          >
            <div
              class="rowElement false"
            >
              <div>
                marwen
              </div>
              <div>
                hanzouli
              </div>
              <div>
                <div
                  class="EmpruntsDetails"
                >
                  <span
                    class="icons "
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 512 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M255.8 112c-80.4 0-143.8 50.6-219.6 133.3-5.5 6.1-5.6 15.2-.1 21.3C101 338.3 158.2 400 255.8 400c96.4 0 168.7-77.7 220.1-134 5.3-5.8 5.6-14.6.5-20.7C424 181.8 351.5 112 255.8 112zm4.4 233.9c-53 2.4-96.6-41.2-94.1-94.1 2.1-46.2 39.5-83.6 85.7-85.7 53-2.4 96.6 41.2 94.1 94.1-2.1 46.2-39.5 83.6-85.7 85.7z"
                      />
                      <path
                        d="M256 209c0-6 1.1-11.7 3.1-16.9-1 0-2-.1-3.1-.1-36.9 0-66.6 31.4-63.8 68.9 2.4 31.3 27.6 56.5 58.9 58.9 37.5 2.8 68.9-26.9 68.9-63.8 0-1.3-.1-2.6-.1-3.9-5.6 2.5-11.7 3.9-18.2 3.9-25.2 0-45.7-21.1-45.7-47z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div>
                <svg
                  class="add iconbani"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zM704 536c0 4.4-3.6 8-8 8H544v152c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V544H328c-4.4 0-8-3.6-8-8v-48c0-4.4 3.6-8 8-8h152V328c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v152h152c4.4 0 8 3.6 8 8v48z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
